pkg/serviceaccounts/controllers/docker_registry_service.go

```diff
@@ -168,20 +255,24 @@ func (e *DockerRegistryServiceController) handleLocationChange(serviceLocation s
 
 		dockercfgMap := map[string]credentialprovider.DockerConfigEntry(*dockercfg)
 		keys := sets.KeySet(reflect.ValueOf(dockercfgMap))
-		if len(keys) != 1 {
+		if len(keys) == 0 {
```

Comment:
> in this case, or in the case where an update below fails on a conflict, remind me how the secret eventually gets cleaned up or fixed? is it on the dockercfgController resync?

Ending up with zero keys can only happen with manual intervention (note the default that's always added).  If that's done, then only manual intervention (deletion) can clean it up.